cardano-launcher/src/Cardano/Shell/Launcher.hs

```diff
@@ -135,3 +151,101 @@ runWallet ed walletPath walletArguments updaterData = do
             , Process.std_err   = stdStream
             }
 
+-- Todo: Add haddock comment for each field
+-- | Launcher options
+data LauncherOptions = LauncherOptions
+    { loConfiguration       :: !ConfigurationOptions
+    , loTlsPath             :: !FilePath
+    , loUpdaterPath         :: !FilePath
+    , loUpdaterArgs         :: ![Text]
+    , loUpdateArchive       :: !(Maybe FilePath)
+    , loUpdateWindowsRunner :: !(Maybe FilePath)
+    , loWalletPath          :: !FilePath
+    , loWalletArgs          :: ![Text]
+    } deriving (Show, Generic)
+
+instance FromJSON LauncherOptions where
+    parseJSON = withObject "LauncherOptions" $ \o -> do
+        walletPath <- o .: "walletPath"
+        walletArgs <- o .: "walletArgs"
+        updaterPath <- o .: "updaterPath"
+        updaterArgs <- o .: "updaterArgs"
+        updateArchive <- o .: "updateArchive"
+        updateWindowsRunner <- o .: "updateWindowsRunner"
+        configuration <- o .: "configuration"
+        tlsPath <- o .: "tlsPath"
+        pure $ LauncherOptions
+            configuration
+            tlsPath
+            updaterPath
+            updaterArgs
+            updateArchive
+            updateWindowsRunner
+            walletPath
+            walletArgs
+
+-- | Configuration yaml file location and the key to use. The file should
+-- parse to a MultiConfiguration and the 'cfoKey' should be one of the keys
+-- in the map.
+data ConfigurationOptions = ConfigurationOptions
```

Comment:
The name is worse than it was.
Revert the name, `ConfigurationOptions` is to `Application` what `CarSignal` is to `Car`. Can we be more specific?